lib/NavigatorWatcher.js

```diff
@@ -68,6 +74,23 @@ class NavigatorWatcher {
     });
   }
 
+  /**
+   * @param {!Puppeteer.Request} request
+   */
+  _onRequest(request) {
+    if (request.frame() !== this._frame || !request.isNavigationRequest())
+      return;
+    this._navigationRequests.set(request.url(), request);
```

Comment:
we do we care about the url here? Shouldn't just the most recent navigation request be the only one that matters?

Reply:
Fair.